array_test.go

```diff
@@ -748,3 +748,41 @@ func TestArrayConvertContains(t *testing.T) {
 	value.chain.assertFailed(t)
 	value.chain.reset()
 }
+
+func TestArrayEvery(t *testing.T) {
+	reporter := newMockReporter(t)
+
+	tableTests := []struct {
+		name     string
+		input    *Array
+		function func(idx int, val *Value)
+		output   *Array
+	}{
+		{
+			name:  "Non-empty array",
+			input: NewArray(reporter, []interface{}{1, 2}),
+			function: func(_ int, val *Value) {
```

Comment:
Nitpick: can we also check that passed index is correct?

Reply:
Sure. I will also modify the tests according to new change.